### PR TITLE
test: finished all ui components test

### LINKED ISSUE #3507: test: Unit test for UI component & code coverage
### Goal
Ensure that the individual parts of your UI function as expected. 

Step
- [X] Setup Jest for Joi folder 

Write Unit Tests Hooks
- [X] useClipboard
- [X] usePageLeave
- [X]  useTextSelection
- [X]  useClickOutside
- [X]  useOs
- [X]  useMediaQuery

Write Unit Tests Components
- [X] Tooltip
- [X] ScrollArea
- [X] Button
- [X] Switch
- [X] Progress
- [X] Checkbox
- [X] Badge
- [X] Modal
- [X] Slider
- [X] Input
- [X] Select
- [X] TextArea
- [X] Tabs
- [X] Accordion

PR
https://github.com/janhq/jan/pull/3540
https://github.com/janhq/jan/pull/3588

### PR DESCRIPTION
**Pull request by urmauur**
## Describe Your Changes

Write Unit Tests Components
- [X] Tooltip
- [X] ScrollArea
- [X] Switch
- [X] Progress
- [X] Checkbox
- [X] Modal
- [X] Slider
- [X] Input
- [X] Select
- [X] TextArea
- [X] Tabs

Finished All UI components test case

## Fixes Issues

- Closes #3507 
- Closes #

## Self Checklist

- [ ] Added relevant comments, esp in complex areas
- [ ] Updated docs (for bug fixes / features)
- [ ] Created issues for follow-up changes or refactoring needed

### CODE REVIEW

**Comment by dan-homebrew** [comment]
@urmauur Lgtm

**Review by dan-homebrew — APPROVED** [review]

File: joi/src/core/Modal/Modal.test.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    expect(screen.queryByLabelText('Close')).not.toBeInTheDocument()
+  })
+
+  it('calls onOpenChange when opening and closing the modal', () => {
+    const onOpenChange = jest.fn()
+    render(
+      <Modal
+        trigger={<button>Open Modal</button>}
+        content={<div>Modal Content</div>}
+        onOpenChange={onOpenChange}
```
Comment (louis-jan) [inline joi/src/core/Modal/Modal.test.tsx:68]: Everything looks good. The CC is increased, yay!

nits
```suggestion
    render(
      <Modal
        trigger={<button>Open Modal</button>}
        content={<div>Modal Content</div>}
        onOpenChange={jest.fn(}
```

**Review by louis-jan — APPROVED** [review]
LGTM, YAY

File: joi/src/core/Modal/Modal.test.tsx (hunk 1 above)
Comment (urmauur) [reply joi/src/core/Modal/Modal.test.tsx]: hy @louis-jan i create a `const` `jest.fn()` for test the function is toHaveBeenCalledWith, but for avoid confused i renaming to `onOpenChangeMock` instead `onOpenChange`

**Pull request merged**